WordPress/gutenberg
# Commands: Try debouncing search for post-type navigation

## What?
PR tries to debounce search requests made by post-type navigation commands.

## Why?
Currently, the command is making a request for each keystroke, which can be expensive depending on the WP installation size.

## Testing Instructions
1. Open a post or page.
2. Open the Command palette.
3. Search for a post or a page by title.
4. Confirm that network requests are delayed.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/240569/8b81ae52-f4e2-4939-ac7d-366b45d97aca
